evm/constants.py

```diff
@@ -152,7 +152,7 @@
 GENESIS_GAS_LIMIT = 3141592
 GENESIS_PARENT_HASH = ZERO_HASH32
 GENESIS_COINBASE = ZERO_ADDRESS
-GENESIS_NONCE = b'\x00\x00\x00\x00\x00\x00\x00*'  # 42 encoded as big-endian-integer
+GENESIS_NONCE = b'\x00\x00\x00\x00\x00\x00\x00B'  # 42 encoded as big-endian-integer
```

Comment:
Why this?  Isn't this `66`?

Reply:
With the original value the hash for the genesis header does not match the expected value (and causes geth to refuse the peer). IIUC, the nonce should be 0x42 encoded as big-endian-int

```
In [4]: encode_hex(MAINNET_GENESIS_HEADER.hash)
Out[4]: '0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3'

In [5]: MAINNET_GENESIS_HEADER.nonce
Out[5]: b'\x00\x00\x00\x00\x00\x00\x00B'

In [6]: MAINNET_GENESIS_HEADER.nonce = b'\x00\x00\x00\x00\x00\x00\x00*'

In [7]: encode_hex(MAINNET_GENESIS_HEADER.hash)
Out[7]: '0x8fcee18008327b9b03f9001a8ad32e66bea58f6694a43df7ad47f388b29bc489'
```

Reply:
Looks good.  Thanks for catching the mistake.